scripts/west_commands/ncs_provision.py

```diff
@@ -164,10 +164,7 @@ def do_add_parser(self, parser_adder):
         upload_parser.add_argument(
             "-s",
             "--soc",
-            type=str,
-            help="SoC",
-            choices=["nrf54l05", "nrf54l10", "nrf54l15", "nrf54lm20a"],
-            required=True,
+            help="Deprecated, is not used anymore."
```

Comment:
https://docs.python.org/3/library/argparse.html#deprecated

Reply:
Added in version 3.13.
so maybe later we could use that feature